src/metric.rs

```diff
@@ -11,6 +11,8 @@
 //! of these components can choose what metrics they’re interested in and also
 //! can add their own custom metrics without the need to maintain forks.
 
+#![cfg(target_has_atomic = "64")]
```

Comment:
This would result in a compilation error when this condition is not true.

To implement this, it should be on https://github.com/rust-vmm/vmm-sys-util/blob/main/src/lib.rs#L21 like:

```rust
#[cfg(target_has_atomic = "64")]
pub mod metric;
```

Reply:
Well, it doesn't result in compilation error: see https://buildd.debian.org/status/package.php?p=rust-vmm-sys-util&suite=sid - it compiled fine on all platforms where previously this caused an error.  I guess it will result in compilation error of crates which *use* this one, as there will be no implementation for the declared functionality.

But I must admit: I know *nothing* about rust. It's the 2nd day since I looked at what rust actually *is*, what I suggest is just something I've seen being used elsewhere. I've no idea if this is a correct solution or not.

Reply:
Writing `#![cfg(target_has_atomic = "64")]` (notice the `!`) is equivalent to:
```rust
#[cfg(target_has_atomic = "64")] pub mod metric;
```

We can still have it as @JonathanWoollett-Light suggested even if they're equivalent because it is making it a bit more obvious that the whole module is excluded from compilation on targets that don't support atomic64.